src/tests/intg/test_confdb.py

```diff
@@ -0,0 +1,151 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import stat
+import signal
+import subprocess
+import time
+import pytest
+
+import config
+from util import unindent
+
+
+def create_conf_file(contents):
+    """Create sssd.conf with specified contents"""
+    conf = open(config.CONF_PATH, "w")
+    conf.write(contents)
+    conf.close()
+    os.chmod(config.CONF_PATH, stat.S_IRUSR | stat.S_IWUSR)
+
+
+def cleanup_conf_file():
+    """Remove sssd.conf, if it exists"""
+    if os.path.lexists(config.CONF_PATH):
+        os.unlink(config.CONF_PATH)
+
+
+def create_conf_cleanup(request):
+    """Add teardown for removing sssd.conf"""
+    request.addfinalizer(cleanup_conf_file)
+
+
+def create_conf_fixture(request, contents):
+    """
+    Create sssd.conf with specified contents and add teardown for removing it
+    """
+    create_conf_file(contents)
+    create_conf_cleanup(request)
+
+
+def create_sssd_process():
+    """Start the SSSD process"""
+    if subprocess.call(["sssd", "-D", "--logger=files"]) != 0:
+        raise Exception("sssd start failed")
+
+
+def get_sssd_pid():
+    pid_file = open(config.PIDFILE_PATH, "r")
+    pid = int(pid_file.read())
+    return pid
+
+
+def cleanup_sssd_process():
+    """Stop the SSSD process and remove its state"""
+    try:
+        pid = get_sssd_pid()
+        os.kill(pid, signal.SIGTERM)
+        while True:
+            try:
+                os.kill(pid, signal.SIGCONT)
+            except OSError:
+                break
+            time.sleep(1)
+    except OSError:
+        pass
+    for path in os.listdir(config.DB_PATH):
+        os.unlink(config.DB_PATH + "/" + path)
+    for path in os.listdir(config.MCACHE_PATH):
+        os.unlink(config.MCACHE_PATH + "/" + path)
+
+
+def test_domains__domains(request):
+    """
+    Test that SSSD starts without domains option.
+    """
+    conf = unindent("""\
+        [sssd]
+        services = nss, sudo
+        domains = test
+
+        [domain/test]
+        id_provider = files
+    """)
+
+    create_conf_fixture(request, conf)
+
+    try:
+        create_sssd_process()
+    except Exception:
+        assert False
+    finally:
+        cleanup_sssd_process(request)
+
+
+def test_domains__enabled(request):
+    """
+    Test that SSSD starts without domains option.
+    """
+    conf = unindent("""\
+        [sssd]
+        services = nss, sudo
+
+        [domain/test]
+        enabled = true
+        id_provider = files
+    """)
+
+    create_conf_fixture(request, conf)
+
+    try:
+        create_sssd_process()
+    except Exception:
+        assert False
+    finally:
+        cleanup_sssd_process(request)
+
+
+def test_domains__empty(request):
+    """
+    Test that SSSD fails without any domain enabled.
+    """
+    conf = unindent("""\
+        [sssd]
+        services = nss, sudo
```

Comment:
I would like to suggest to add `enable_files_domain = false` explicitly to cover the different default we currently have on different platforms to make sure the implicit files provider is not started accidentally.